torch/jit/annotations.py

```diff
@@ -75,11 +75,44 @@ def get_signature(fn, _n_arguments=None, _n_binders=None):
     # This might happen both because we failed to get the source of fn, or
     # because it didn't have any annotations.
     if type_line is None:
-        return default_signature(fn, source, _n_arguments, _n_binders)
+        return None
 
     return parse_type_line(type_line)
 
 
+# This is essentially a weaker form of get_signature(), where we don't care if
+# we have the types, we just care that we can figure out how many parameters
+# a function takes.
+def get_num_params(fn):
```

Comment:
When do we need to know this?

Reply:
I removed the concept of returning a default signature from this file, because it was conflating too many concepts behind the `get_signature` API. Now it's the caller's responsibility to create a default signature